code/game/machinery/vending.dm

```diff
@@ -94,6 +94,9 @@
 		build_inventory(premium, 0, 1, start_empty = 1)
 		for(var/obj/item/weapon/vending_refill/VR in component_parts)
 			refill_inventory(VR, product_records)
+		for(var/obj/item/weapon/vending_refill/VR in component_parts)
+			refill_inventory(VR, coin_records)
+			refill_inventory(VR, hidden_records)

```

Comment:
This is so that the refills are first used to replenish the standard products, and then try to refill the hidden stuff. It's to prevent abusing the machine by reconstructing it multiple times to get more hidden products.